src/encoded/static/components/auditmatrixv1.js

```diff
@@ -0,0 +1,740 @@
+        super(props);
+
+        this.state = {
+            /** Categories the user has expanded */
+            expandedRowCategories: [],
+            /** True if matrix scrolled all the way to the right; used for flashing arrow */
+            scrolledRight: false,
+        };
+        this.expanderClickHandler = this.expanderClickHandler.bind(this);
+        this.handleOnScroll = this.handleOnScroll.bind(this);
+        this.handleScrollIndicator = this.handleScrollIndicator.bind(this);
+    }
+
+    componentDidMount() {
+        this.handleScrollIndicator(this.scrollElement);
+    }
+
+    componentDidUpdate(prevProps) {
+        // If URI changed, we need close any expanded rowCategories in case the URI change results
+        // in a huge increase in displayed data. Also update the scroll indicator if needed.
+        if (prevProps.context['@id'] !== this.props.context['@id']) {
+            this.handleScrollIndicator(this.scrollElement);
+            this.setState({ expandedRowCategories: [] });
+        }
+    }
+
+    /**
+     * Called when the user clicks on the expander button on a category to collapse or expand it.
+     * @param {string} category Key for the category
+     */
+    expanderClickHandler(category) {
+        this.setState((prevState) => {
+            const matchingCategoryIndex = prevState.expandedRowCategories.indexOf(category);
+            if (matchingCategoryIndex === -1) {
+                // Category doesn't exist in array, so add it.
+                return { expandedRowCategories: prevState.expandedRowCategories.concat(category) };
+            }
+
+            // Category does exist in array
+            // Move close to header
+            const header = document.querySelector(`#${sanitizeId(category)}`);
+            const headerToPageTopDistance = header ? header.getBoundingClientRect().top : 0;
+            const buffer = 20; // extra space between navbar and header
+            const top = headerToPageTopDistance - (getNavbarHeight() + buffer);
+            window.scrollBy({
+                top,
+                left: 0,
+                behavior: 'smooth',
+            });
+
+            // Remove category.
+            const expandedCategories = prevState.expandedRowCategories;
+            return { expandedRowCategories: [...expandedCategories.slice(0, matchingCategoryIndex), ...expandedCategories.slice(matchingCategoryIndex + 1)] };
+        });
+    }
+    /**
+     * Called when the user scrolls the matrix horizontally within its div to handle scroll
+     * indicators.
+     * @param {object} e React synthetic scroll event
+     */
+    handleOnScroll(e) {
+        this.handleScrollIndicator(e.target);
+    }
+
+    /**
+     * Show a scroll indicator depending on current scrolled position.
+     * @param {object} element DOM element to apply shading to
+     */
+    handleScrollIndicator(element) {
+        // Have to use a "roughly equal to" test because of an MS Edge bug mentioned here:
```

Comment:
Is this still important in that Microsoft is dumping trident-based MS Edge with a chromium-based Edge?